### ISSUE TITLE
Error Obtaining ID Token in Build and Publish Documentation Workflow

### ISSUE THREAD

**Issue by dzuluaga** [openmobilehub/android-omh-auth#72]
Hello Dominik and Natalia,

I recently encountered an issue while trying to run the `Build and Publish Documentation` workflow in the OMH Auth repository. During the deployment phase, the workflow failed with an error related to obtaining an ID token. The error message indicated a missing `ACTIONS_ID_TOKEN_REQUEST_URL` environment variable and suggested that the `GITHUB_TOKEN` did not have the `id-token: write` permission.

**Error Details:**
```
Run actions/deploy-pages@v2
Error: Error message: Unable to get ACTIONS_ID_TOKEN_REQUEST_URL env variable
    at Function.<anonymous> (/home/runner/work/_actions/actions/deploy-pages/v2/node_modules/@actions/core/lib/oidc-utils.js:71:1)
    at Generator.next (<anonymous>)
    at /home/runner/work/_actions/actions/deploy-pages/v2/node_modules/@actions/core/lib/oidc-utils.js:8:1
    at new Promise (<anonymous>)
    at __webpack_modules__.8041.__awaiter (/home/runner/work/_actions/actions/deploy-pages/v2/node_modules/@actions/core/lib/oidc-utils.js:4:1)
    at Function.getIDToken (/home/runner/work/_actions/actions/deploy-pages/v2/node_modules/@actions/core/lib/oidc-utils.js:57:1)
    at Object.<anonymous> (/home/runner/work/_actions/actions/deploy-pages/v2/node_modules/@actions/core/lib/core.js:315:1)
    at Generator.next (<anonymous>)
    at /home/runner/work/_actions/actions/deploy-pages/v2/node_modules/@actions/core/lib/core.js:27:1
    at new Promise (<anonymous>)
Error: Ensure GITHUB_TOKEN has permission "id-token: write".
```

**Link to the Failed Job:**
[Failed Job Log](https://github.com/openmobilehub/android-omh-auth/actions/runs/8822973936/job/24222438119)

It appears that the workflow might be missing the necessary permissions for the `GITHUB_TOKEN` specifically around `id-token: write`. This is crucial for deploying the documentation to GitHub Pages using `actions/deploy-pages@v2`.

**Suggested Fix:**
To resolve this issue, we could add the following permissions block either at the job level (if other jobs do not need these permissions) or at the workflow level (to cover all jobs):

```yaml
permissions:
  contents: read
  id-token: write
```

This change should provide the `GITHUB_TOKEN` with the required permissions to obtain an ID token, thus preventing the error from occurring.

Could we please make sure that changes like these are tested in a non-main branch before merging to ensure that the deployment process remains stable and predictable? It would help prevent such disruptions and maintain the reliability of our CI/CD pipeline.

Thank you for looking into this!

Please ensure OMH Maps also reflect these changes once the solution is tested.

CC - @prestonlau

**Comment by adamTrz** [comment]
Hey @dzuluaga - mind checking it again please? Should be fixed with #74  :)

**Comment by dzuluaga** [comment]
@adamTrz - The job runs okay. But the images are missing. Am I missing anything?

<img width="1505" alt="image" src="https://github.com/openmobilehub/android-omh-auth/assets/1755383/af6ef027-3e70-41a9-8eb0-0bd29312fad4">
